src/util/simplify_expr2.cpp

```diff
@@ -415,13 +414,62 @@ static type2tc common_arith_op2_type(expr2tc &e, expr2tc &f)
 
 expr2tc add2t::do_simplify() const
 {
+  // (A + 1) + ~B --> A - B
+  if(is_add2t(side_1) && is_bitnot2t(side_2))
+  {
+    auto sidecheck = to_add2t(side_1).side_2;
+    auto sidecheck2 = to_add2t(side_1).side_1;
+    if(to_constant_int(sidecheck).value == 1)
+    {
+      auto B = to_bitnot2t(side_2).value;
+      auto new_operand = sub2tc(side_1->type, sidecheck2, B);      
+      return new_operand;
+    }
+  }
+  // (~B + A) + 1 --> A - B
+  //&&
+  // (A + ~B) + 1 --> A - B
+  if(is_add2t(side_1) && to_constant_int(side_2).value == 1)
```

Comment:
You can transform this into a lambda function, then you will be able to handle the commutative operator